src/mesh/mesh_adaptation.cpp

```diff
@@ -0,0 +1,119 @@
+#include "mesh_adaptation.h"
+
+namespace PHiLiP {
+
+template <int dim, typename real, typename MeshType>
+MeshAdaptation<dim,real,MeshType>::MeshAdaptation(double critical_res_input, int total_ref_cycle, double refine_frac, double coarsen_frac)
+    : critical_residual(critical_res_input)
+    , total_refinement_cycles(total_ref_cycle)
+    , refinement_fraction(refine_frac)
+    , coarsening_fraction(coarsen_frac)
+    , current_refinement_cycle(0)
+    {}
+
+template <int dim, typename real, typename MeshType>
+int MeshAdaptation<dim,real,MeshType>::adapt_mesh(std::shared_ptr< DGBase<dim, real, MeshType> > dg)
+{
+    if (!refine_mesh)
+    {
+        return 0;
+    }
+
+    if (total_refinement_cycles == current_refinement_cycle)
+    {
+        return 0;
+    }
+
+    cellwise_errors.reinit(0);
+    cellwise_errors.reinit(dg->high_order_grid->triangulation->n_active_cells());
+    compute_cellwise_errors(dg);
+
+    fixed_fraction_isotropic_refinement_and_coarsening(dg);
+    current_refinement_cycle++;
+    std::cout<<"Refined"<<std::endl;
```

Comment:
This will get outputted by every processor. Look into the code for other uses of pcout where only the first processor outputs this message.

Reply:
Thanks, it now uses pcout.